pkg/upgrader/upgrader_v1alpha1.go

```diff
@@ -341,8 +341,8 @@ func waitForHealthy(node corev1.Node, config *restclient.Config) (err error) {
 	return nil
 }
 
-func getVersion(ctx context.Context, client *client.Client) (version *machineapi.VersionInfo, err error) {
-	err = retry.Constant(15*time.Minute, retry.WithUnits(3*time.Second), retry.WithJitter(500*time.Millisecond)).Retry(func() error {
+func getVersion(ctx context.Context, client *client.Client, timeout time.Duration) (version *machineapi.VersionInfo, err error) {
```

Comment:
I'd like to maybe make this configurable but the CRD, but that is for another time. Probably only makes sense to make it configurable when we need to verify a node's version after upgrading. It may vary from machine type to machine type.